server/api/smallgroup/smallgroup.controller.js

```diff
@@ -304,5 +304,20 @@ function generateCode(codeLength){
       var character = (Math.floor(Math.random() * characterOptions.length));
       code = code.concat(characterOptions[character.toString()]);
   }
+  ClassYear.findOne({"dayCodes.code":code})
+    .exec(function(err, classYear){
+      if (err) return handleError(res, err);
+      if(classYear) {
+        generateCode(codeLength);
+      }
+      else{
+        SmallGroup.findOne({"dayCodes.code":code})
+          .exec(function(err, smallgroup){
+            if (err) return handleError(res, err);
+            if(smallgroup){
+              generateCode(codeLength);
+            }
+        });}
```

Comment:
should probably bump the `}` to the next line to align with the else indentation